plugins/core-classes/src/main/java/org/qbicc/plugin/coreclasses/ArrayLengthBasicBlockBuilder.java

```diff
@@ -0,0 +1,26 @@
+package org.qbicc.plugin.coreclasses;
+
+import org.qbicc.context.CompilationContext;
+import org.qbicc.graph.BasicBlockBuilder;
+import org.qbicc.graph.DelegatingBasicBlockBuilder;
+import org.qbicc.graph.ValueHandle;
+import org.qbicc.type.ArrayObjectType;
+import org.qbicc.type.ValueType;
+
+public class ArrayLengthBasicBlockBuilder extends DelegatingBasicBlockBuilder {
+    private final CompilationContext ctxt;
+
+    public ArrayLengthBasicBlockBuilder(CompilationContext ctxt, BasicBlockBuilder delegate) {
+        super(delegate);
+        this.ctxt = ctxt;
+    }
+
+    @Override
+    public ValueHandle lengthOf(ValueHandle array) {
```

Comment:
There is exactly one use of `lengthOf`...from MethodParser because it can't depend on CoreClasses.